validation/mock_trace_validation.go

```diff
@@ -0,0 +1,68 @@
+package validation
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/golang/protobuf/ptypes"
+	"google.golang.org/genproto/googleapis/devtools/cloudtrace/v2"
+)
+
+const (
+	InvalidTimestampMsg   = "start time must be before end time"
+	MalformedTimestampMsg = "unable to parse timestamp"
+	MissingFieldMsg       = "span is missing required fields: %v"
+)
+
+var (
+	requiredMessageFields = []string{"DisplayName", "StartTime", "EndTime"}
+	requiredStringFields  = []string{"Name", "SpanId"}
+)
+
+func IsSpanValid(span *cloudtrace.Span) (bool, string) {
```

Comment:
Just return `error` instead of (bool, string). Its clear enough that nil means success.
```func ValidateSpan(span *cloudtrace.Span) error {```

Reply:
Done